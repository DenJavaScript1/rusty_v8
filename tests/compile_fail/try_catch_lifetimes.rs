extern crate rusty_v8 as v8;

pub fn main() {
  let context: v8::Local<v8::Context> = mock();
  let scope: &mut v8::scope::Entered<'_, v8::HandleScope> = mock();

  let _leaked = {
    let mut try_catch = v8::TryCatch::new(scope);
    let tc = try_catch.enter();
    let exception = tc.exception().unwrap();
    let stack_trace = tc.stack_trace(scope, context).unwrap();
    let message = tc.message().unwrap();
    (exception, stack_trace, message)
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
