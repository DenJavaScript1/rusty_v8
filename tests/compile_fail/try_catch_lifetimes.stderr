error[E0597]: `try_catch` does not live long enough
  --> $DIR/try_catch_lifetimes.rs:9:14
   |
7  |   let _leaked = {
   |       ------- borrow later stored here
8  |     let mut try_catch = v8::TryCatch::new(scope);
9  |     let tc = try_catch.enter();
   |              ^^^^^^^^^ borrowed value does not live long enough
...
14 |   };
   |   - `try_catch` dropped here while still borrowed
